src/editor/StyleEditor.cpp

```diff
@@ -326,4 +353,117 @@ QStringList getStyles() {
   return styles;
 }
 
+#if defined(Q_OS_WINDOWS)
+void customizeNativeTitleBar(WId w) noexcept {
+  if (QOperatingSystemVersion::current() >=
+      QOperatingSystemVersion::Windows10) {
+    static ULONG osBuildNumber = *reinterpret_cast<DWORD *>(0x7FFE0000 + 0x260);
+    // https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/ntddk/ns-ntddk-kuser_shared_data#syntax
+    // member "NtMinorVersion" of _KUSER_SHARED_DATA. don't even ask where the
+    // address comes from
+
+    auto qColorToColorRef = [](const QColor &c) -> COLORREF {
+      return (c.red() | c.green() << 8 | c.blue() << 16);
+    };
+
+    enum {
+      NoDwm,
+      NoDwmSetWindowAttribute,
+      NoDarkMode,
+      NoBorderColor,
+      NoCaptionColor,
+      NoTextColor
+    };
+
+    static auto error = [](int i) {
+      static QHash<const int, const char *> errorTable = {
+          {NoDwm, "DWM library not found."},
+          {NoDwmSetWindowAttribute, "DwmSetWindowAttribute not found."},
+          {NoDarkMode, "Unable to set immersive dark mode attribute."},
+          {NoBorderColor, "Unable to set window border color attribute."},
+          {NoCaptionColor, "Unable to set window caption color attribute."},
+          {NoTextColor, "Unable to set window text color attribute."},
+      };
+      static QVector<int> alreadyErrored;
+      if (!alreadyErrored.contains(i)) {
+        qWarning() << "Error while setting title bar attributes: "
+                   << QObject::tr(errorTable.find(i).value());
+        alreadyErrored.push_back(i);
+      }
+    };
+
+    static auto hdwmapi = LoadLibraryW(L"dwmapi.dll");
+    if (!hdwmapi) {
+      error(NoDwm);
+      return;
+    }
+
+    //  https://learn.microsoft.com/en-us/windows/win32/api/dwmapi/ne-dwmapi-dwmwindowattribute
+    typedef int(WINAPI * DWMSETWINDOWATTRIBUTE)(
+        HWND hwnd, DWORD dwAttribute, LPCVOID pvAttribute,
+        DWORD cbAttribute);  // typedef is required here thanks to WINAPI
+                             // keyword (__stdcall) >:|
+    static auto DwmSetWindowAttribute = reinterpret_cast<DWMSETWINDOWATTRIBUTE>(
+        GetProcAddress(hdwmapi, "DwmSetWindowAttribute"));
+    if (!DwmSetWindowAttribute) {
+      error(NoDwmSetWindowAttribute);
+      return;
+    }
+
+    auto hwnd = reinterpret_cast<HWND>(w);
+
+    quint32 darkTitleBar = StyleEditor::config.other.Win10BorderDark;
+    if (FAILED(DwmSetWindowAttribute(
+            hwnd,
+            (osBuildNumber >= 18985)
```

Comment:
howd you get this build number? nad the 22k below

Reply:
they are windows build numbers. first release of win11 is 10.0.22000 (major, minor, build). they are constant